src/main/java/org/thoughtcrime/securesms/ConversationListActivity.java

```diff
@@ -287,7 +287,10 @@ public void refreshUnreadIndicator() {
     int skipId = dcAccounts.getSelectedAccount().getAccountId();
     for (int accountId : dcAccounts.getAll()) {
       if (accountId != skipId) {
-        unreadCount += dcAccounts.getAccount(accountId).getFreshMsgs().length;
+        DcContext dcContext = dcAccounts.getAccount(accountId);
+        if (!dcContext.isMuted()) {
+          unreadCount += dcContext.getFreshMsgs().length;
+        }
```

Comment:
this is about the badge counter in the upper left corner right?

i think, it is not correct to leave muted contexts out, see comments in the core-pr.

we also do not hide these counters for muted-chats.

instead, we should grey-out the counter if there are _only_ muted messages.

for the global badge-counter or unread-dot on the app-icon, i am unsure, however :)

Reply:
> for the global badge-counter or unread-dot on the app-icon

I vote for exclude, because muted chats are also excluded. (at least on desktop and in every messaging app I know that supports muting chats).

I understand and agree it makes some sense to have an indicator of "there are new messages in muted" accounts in the app, though.

Reply:
> we also do not hide these counters for muted-chats.

what??? I think we already do, it is done automatically by core, muted chats don't count in the unread messages count of an account 

similarly it feels natural that if you mute an account you don't want it to keep making noise in your badge counter and then you go check and it turned out it was just a muted account, if you want them to appear there then just don't mute the account

Reply:
> instead, we should grey-out the counter if there are _only_ muted messages.

mmmh it seems there is a misunderstanding, the account badge is already shown in gray for muted accounts, what these lines of code do is gather all the badge counts of all accounts and aum it but muted accounts are left out, trying to be smart here and show the badge counter in gray if all the messages are from muted accounts it is tricky, since you can have mixed counters from muted and not muted accounts and then it is confusing

Reply:
if it is muted+unmuted, i'd use normal color

Reply:
k, after internal discussions, we leave the muted accounts out of the sum-up badge counter in the upper left corner.

one can argue for both, and then, it is easier to not introduce another color/state there :) also, this makes it easier wrt global badge counter. and we can always iterate :)